Makefile

```diff
@@ -0,0 +1,55 @@
+DOCKER    = docker
+
+BASE_REPO = cloudflare/quiche
+BASE_TAG  = latest
+
+QNS_REPO  = cloudflare/quiche-qns
+QNS_TAG   = latest
+
+FUZZ_REPO = kestrel.forallsecure.com:5000/ghedo/quiche-libfuzzer
+FUZZ_TAG  = latest
+
+docker-build: docker-base docker-qns
+
+# build quiche-apps only
+.PHONY: build-apps
+build-apps:
+	cargo build --manifest-path tools/apps/Cargo.toml
+
+# build base image
+.PHONY: docker-base
+docker-base: Dockerfile
+	$(DOCKER) build --target quiche-base -t $(BASE_REPO):$(BASE_TAG) .
+
+# build qns image
+.PHONY: docker-qns
+docker-qns: Dockerfile tools/qns/run_endpoint.sh
+	$(DOCKER) build --target quiche-qns -t $(QNS_REPO):$(QNS_TAG) .
+
+.PHONY: docker-publish
+docker-publish:
+	$(DOCKER) push $(BASE_REPO):$(BASE_TAG)
+	$(DOCKER) push $(QNS_REPO):$(QNS_TAG)
+
+# build fuzzers
```

Comment:
When will those fuzz targets run?

Reply:
I run these manually for now, since we don't really want to stop the fuzz runs for every commit.